dist/init/freebsd/caddy

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+#
+
+# PROVIDE: caddy
+# REQUIRE: LOGIN NETWORKING named cleanvar sshd
+# KEYWORD: shutdown
+
+#
+# Add the following lines to /etc/rc.conf to enable caddy:
+# caddy_enable (bool):        Set to "NO" by default.
+#                             Set it to "YES" to enable caddy
+#
+# caddy_cert_email (str):     Set to "" by default.
+#                             Defines the SSL certificate issuer email. By providing an
+#                             email address you automatically agree to letsencrypt.org's
+#                             general terms and conditions
+#
+# caddy_bin_path (str):       Set to "/usr/local/bin/caddy" by default.
+#                             Provides the path to the caddy server executable
+#
+# caddy_cpu (str):            Set to "99%" by default.
+#                             Configures, how much CPU capacity caddy may gain
+#
+# caddy_config_path (str):    Set to "/usr/local/www/Caddyfile" by default.
+#                             Defines the path for the configuration file caddy will load on boot
+#
+# caddy_run_user (str):       Set to "root" by default.
+#                             Defines the user that caddy will run on
+#
+
+. /etc/rc.subr
+
+name="caddy"
+rcvar=${name}_enable
+
+load_rc_config $name
+: ${caddy_enable:=no}
+: ${caddy_cert_email=""}
+: ${caddy_bin_path="/usr/local/bin/caddy"}
+: ${caddy_cpu="99%"} # was a bug for me that caused a crash within jails
```

Comment:
@mholt If you're talking about the `load_rc_config` block in general: FreeBSD's rc.d has a central configuration file called `rc.conf` where every startup service is configured. `caddy_enable` is manditory by the system and is the variable that enables caddy on boot (as there are no runlevels).  `caddy_cert_email` is not required by FreeBSD itself but must be filled - otherwise Caddy will switch to interactive mode on startup and this is not good for running as a daemon. 

If you're curious about the `caddy_cpu="99%"`: I had issues running Caddy in FreeBSD jails (which almost everyone does on FreeBSD) when cpu caps were set to 100%. After some weeks I had a process not responding to HTTP requests and not reacting to any kind of signals - even not `SIGKILL`. The only way to get rid of it was to restart the machine. But as I was not able to reproduce it as this was my production environment and I wanted to have it fixed as soon as possible, I didn't file a bug report - but I will if I get time to reproduce it within a test environment to provide enough information to fix it.